migrateOVN.sh

```diff
@@ -217,23 +231,34 @@ fi
 nmcli conn mod ovs-if-br-ex 802-3-ethernet.mtu ${iface_mtu} 802-3-ethernet.cloned-mac-address ${iface_mac} \
   ipv4.never-default yes ipv6.never-default yes
 
-# bring down br-ex, bring back up br-ex, bring up old iface
-nmcli conn down br-ex
-nmcli conn up ovs-if-phys0
+# restart NM to auto-activate connections as modified
+systemctl restart NetworkManager
+# give NM breathing time to settle
+sleep 5
 
+# make sure ovs-if-phys0 is active for ${iface} is added to br-ex
+if ! wait_conn ovs-if-phys0; then
+  echo "Timeout waiting for ovs-if-phys0 to activate"
+  exit 1
+fi
 # before reconnecting we need to update the flow in OVS so that DHCP request can get out
-sleep 5
 ofport=$(ovs-vsctl --columns ofport  --bare find interface name=${iface})
 if [ -z "$ofport" ]; then
   echo "Unable to identify OpenFlow port number for interface: ${iface}"
   exit 1
 fi
 ovs-ofctl add-flow br-ex "table=0,priority=101,in_port=LOCAL,actions=output:${ofport}"
-nmcli conn up ovs-if-br-ex
```

Comment:
No longer needed due to NM restart?

Reply:
Yes, not really needed, I had the idea that it might speed things up but I don't want to risk NM doing anything weird.

By default, NM should use dhclient and this is what it says about re-transmissions:
```
initial-interval time;

The initial-interval statement sets the amount of time between the first attempt to reach a server and the second attempt to reach a server. Each time a message is sent, the interval between messages is incremented by twice the current interval multiplied by a random number between zero and one. If it is greater than the backoff-cutoff amount, it is set to that amount. It defaults to ten seconds.

backoff-cutoff time;

The client uses an exponential backoff algorithm with some randomness, so that if many clients try to configure themselves at the same time, they will not make their requests in lockstep.  The backoff-cutoff statement determines the maximum amount of time that the client is allowed to back off, the  actual value will be evaluated randomly between 1/2 to 1 1/2 times the time specified.  It defaults to fifteen seconds.
```
NM itself timeouts dhcp after 45 secs:
```
       dhcp-timeout
           A timeout for a DHCP transaction in seconds. If zero (the default), a globally configured default is used. If still unspecified, a device specific timeout is used (usually 45 seconds). Set to 2147483647 (MAXINT32) for infinity.
```

new log, I only see 1 sec delay tops
```
-- Logs begin at Wed 2021-09-01 15:50:24 UTC, end at Thu 2021-09-02 08:42:13 UTC. --
Sep 01 17:08:10 master-0-1 systemd[1]: Starting Migrates OVS configuration to use a new interface on the host...
Sep 01 17:08:10 master-0-1 migrateOVN.sh[3437]: + old_iface=bond0
Sep 01 17:08:10 master-0-1 migrateOVN.sh[3437]: + iface=bond0.373
Sep 01 17:08:10 master-0-1 migrateOVN.sh[3437]: + trap 'revert 2' ERR
Sep 01 17:08:10 master-0-1 migrateOVN.sh[3437]: + trap 'revert $?' EXIT
Sep 01 17:08:10 master-0-1 migrateOVN.sh[3437]: + for port in ${old_iface} ${iface}
Sep 01 17:08:10 master-0-1 migrateOVN.sh[3437]: + echo 'NIC: bond0 being brought up on this node'
Sep 01 17:08:10 master-0-1 migrateOVN.sh[3437]: NIC: bond0 being brought up on this node
Sep 01 17:08:10 master-0-1 migrateOVN.sh[3437]: + find_nm_conn_for_device bond0
Sep 01 17:08:10 master-0-1 migrateOVN.sh[3437]: + local new_device
Sep 01 17:08:10 master-0-1 migrateOVN.sh[3437]: ++ sed 1d
Sep 01 17:08:10 master-0-1 migrateOVN.sh[3437]: ++ nmcli --fields UUID conn show
Sep 01 17:08:10 master-0-1 migrateOVN.sh[3437]: + conns='6fa43a6d-63b6-455c-b5f0-373ab8fdb460
Sep 01 17:08:10 master-0-1 migrateOVN.sh[3437]: 25456a6d-bf55-3cc5-bfa7-ac1900fd01fa
Sep 01 17:08:10 master-0-1 migrateOVN.sh[3437]: a100e09e-31ad-345e-894c-772a0676719f
Sep 01 17:08:10 master-0-1 migrateOVN.sh[3437]: 87806fe4-bec8-4826-8eca-c715d68cbf24
Sep 01 17:08:10 master-0-1 migrateOVN.sh[3437]: 76682b3f-bbe7-361b-99d0-e1dd5d7b5f6c
Sep 01 17:08:10 master-0-1 migrateOVN.sh[3437]: 937b10f3-05c3-3e07-bad6-0438b0241885
Sep 01 17:08:10 master-0-1 migrateOVN.sh[3437]: 2f9e65a4-3105-4910-86e9-04f3ddbfde79
Sep 01 17:08:10 master-0-1 migrateOVN.sh[3437]: 328d7426-4cf7-4df1-a2ff-61bdb057f653
Sep 01 17:08:10 master-0-1 migrateOVN.sh[3437]: 5dc2327c-9420-49f5-a1f7-1924f359987b
Sep 01 17:08:10 master-0-1 migrateOVN.sh[3437]: 4ddf3d3a-db51-362c-a691-d4b49b53a4de '
Sep 01 17:08:10 master-0-1 migrateOVN.sh[3437]: + IFS=
Sep 01 17:08:10 master-0-1 migrateOVN.sh[3437]: + read -r conn
Sep 01 17:08:10 master-0-1 migrateOVN.sh[3437]: ++ nmcli --get-values connection.interface-name conn show 6fa43a6d-63b6-455c-b5f0-373ab8fdb460
Sep 01 17:08:10 master-0-1 migrateOVN.sh[3437]: + new_device=br-ex
Sep 01 17:08:10 master-0-1 migrateOVN.sh[3437]: + '[' br-ex = bond0 ']'
Sep 01 17:08:10 master-0-1 migrateOVN.sh[3437]: + IFS=
Sep 01 17:08:10 master-0-1 migrateOVN.sh[3437]: + read -r conn
Sep 01 17:08:10 master-0-1 migrateOVN.sh[3437]: ++ nmcli --get-values connection.interface-name conn show 25456a6d-bf55-3cc5-bfa7-ac1900fd01fa
Sep 01 17:08:10 master-0-1 migrateOVN.sh[3437]: + new_device=enp0s3
Sep 01 17:08:10 master-0-1 migrateOVN.sh[3437]: + '[' enp0s3 = bond0 ']'
Sep 01 17:08:10 master-0-1 migrateOVN.sh[3437]: + IFS=
Sep 01 17:08:10 master-0-1 migrateOVN.sh[3437]: + read -r conn
Sep 01 17:08:10 master-0-1 migrateOVN.sh[3437]: ++ nmcli --get-values connection.interface-name conn show a100e09e-31ad-345e-894c-772a0676719f
Sep 01 17:08:10 master-0-1 migrateOVN.sh[3437]: + new_device=bond0.373
Sep 01 17:08:10 master-0-1 migrateOVN.sh[3437]: + '[' bond0.373 = bond0 ']'
Sep 01 17:08:10 master-0-1 migrateOVN.sh[3437]: + IFS=
Sep 01 17:08:10 master-0-1 migrateOVN.sh[3437]: + read -r conn
Sep 01 17:08:10 master-0-1 migrateOVN.sh[3437]: ++ nmcli --get-values connection.interface-name conn show 87806fe4-bec8-4826-8eca-c715d68cbf24
Sep 01 17:08:10 master-0-1 migrateOVN.sh[3437]: + new_device=br-ex
Sep 01 17:08:10 master-0-1 migrateOVN.sh[3437]: + '[' br-ex = bond0 ']'
Sep 01 17:08:10 master-0-1 migrateOVN.sh[3437]: + IFS=
Sep 01 17:08:10 master-0-1 migrateOVN.sh[3437]: + read -r conn
Sep 01 17:08:10 master-0-1 migrateOVN.sh[3437]: ++ nmcli --get-values connection.interface-name conn show 76682b3f-bbe7-361b-99d0-e1dd5d7b5f6c
Sep 01 17:08:10 master-0-1 migrateOVN.sh[3437]: + new_device=enp0s4
Sep 01 17:08:10 master-0-1 migrateOVN.sh[3437]: + '[' enp0s4 = bond0 ']'
Sep 01 17:08:10 master-0-1 migrateOVN.sh[3437]: + IFS=
Sep 01 17:08:10 master-0-1 migrateOVN.sh[3437]: + read -r conn
Sep 01 17:08:10 master-0-1 migrateOVN.sh[3437]: ++ nmcli --get-values connection.interface-name conn show 937b10f3-05c3-3e07-bad6-0438b0241885
Sep 01 17:08:10 master-0-1 migrateOVN.sh[3437]: + new_device=enp0s5
Sep 01 17:08:10 master-0-1 migrateOVN.sh[3437]: + '[' enp0s5 = bond0 ']'
Sep 01 17:08:10 master-0-1 migrateOVN.sh[3437]: + IFS=
Sep 01 17:08:10 master-0-1 migrateOVN.sh[3437]: + read -r conn
Sep 01 17:08:10 master-0-1 migrateOVN.sh[3437]: ++ nmcli --get-values connection.interface-name conn show 2f9e65a4-3105-4910-86e9-04f3ddbfde79
Sep 01 17:08:10 master-0-1 migrateOVN.sh[3437]: + new_device=bond0
Sep 01 17:08:10 master-0-1 migrateOVN.sh[3437]: + '[' bond0 = bond0 ']'
Sep 01 17:08:10 master-0-1 migrateOVN.sh[3437]: + echo 'Detected connection 2f9e65a4-3105-4910-86e9-04f3ddbfde79  for device bond0'
Sep 01 17:08:10 master-0-1 migrateOVN.sh[3437]: Detected connection 2f9e65a4-3105-4910-86e9-04f3ddbfde79  for device bond0
Sep 01 17:08:10 master-0-1 migrateOVN.sh[3437]: ++ echo 2f9e65a4-3105-4910-86e9-04f3ddbfde79
Sep 01 17:08:10 master-0-1 migrateOVN.sh[3437]: ++ tr -d '[:space:]'
Sep 01 17:08:10 master-0-1 migrateOVN.sh[3437]: + detected_conn=2f9e65a4-3105-4910-86e9-04f3ddbfde79
Sep 01 17:08:10 master-0-1 migrateOVN.sh[3437]: + return
Sep 01 17:08:10 master-0-1 migrateOVN.sh[3437]: + echo 'Bringing up connection 2f9e65a4-3105-4910-86e9-04f3ddbfde79'
Sep 01 17:08:10 master-0-1 migrateOVN.sh[3437]: Bringing up connection 2f9e65a4-3105-4910-86e9-04f3ddbfde79
Sep 01 17:08:10 master-0-1 migrateOVN.sh[3437]: + retries=5
Sep 01 17:08:10 master-0-1 migrateOVN.sh[3437]: + i=1
Sep 01 17:08:10 master-0-1 migrateOVN.sh[3437]: + '[' 1 -le 5 ']'
Sep 01 17:08:10 master-0-1 migrateOVN.sh[3437]: + nmcli conn down 2f9e65a4-3105-4910-86e9-04f3ddbfde79
Sep 01 17:08:10 master-0-1 migrateOVN.sh[3437]: Connection 'ovs-if-phys0' successfully deactivated (D-Bus active path: /org/freedesktop/NetworkManager/ActiveConnection/7)
Sep 01 17:08:10 master-0-1 migrateOVN.sh[3437]: + echo 'Brought down connection 2f9e65a4-3105-4910-86e9-04f3ddbfde79'
Sep 01 17:08:10 master-0-1 migrateOVN.sh[3437]: Brought down connection 2f9e65a4-3105-4910-86e9-04f3ddbfde79
Sep 01 17:08:10 master-0-1 migrateOVN.sh[3437]: + nmcli conn up 2f9e65a4-3105-4910-86e9-04f3ddbfde79
Sep 01 17:08:11 master-0-1 migrateOVN.sh[3437]: Connection successfully activated (master waiting for slaves) (D-Bus active path: /org/freedesktop/NetworkManager/ActiveConnection/13)
Sep 01 17:08:11 master-0-1 migrateOVN.sh[3437]: + echo 'Connection 2f9e65a4-3105-4910-86e9-04f3ddbfde79 up successfully'
Sep 01 17:08:11 master-0-1 migrateOVN.sh[3437]: Connection 2f9e65a4-3105-4910-86e9-04f3ddbfde79 up successfully
Sep 01 17:08:11 master-0-1 migrateOVN.sh[3437]: + break
Sep 01 17:08:11 master-0-1 migrateOVN.sh[3437]: + '[' 1 -gt 5 ']'
Sep 01 17:08:11 master-0-1 migrateOVN.sh[3437]: + sleep 10
Sep 01 17:08:21 master-0-1 migrateOVN.sh[3437]: + for port in ${old_iface} ${iface}
Sep 01 17:08:21 master-0-1 migrateOVN.sh[3437]: + echo 'NIC: bond0.373 being brought up on this node'
Sep 01 17:08:21 master-0-1 migrateOVN.sh[3437]: NIC: bond0.373 being brought up on this node
Sep 01 17:08:21 master-0-1 migrateOVN.sh[3437]: + find_nm_conn_for_device bond0.373
Sep 01 17:08:21 master-0-1 migrateOVN.sh[3437]: + local new_device
Sep 01 17:08:21 master-0-1 migrateOVN.sh[3437]: ++ nmcli --fields UUID conn show
Sep 01 17:08:21 master-0-1 migrateOVN.sh[3437]: ++ sed 1d
Sep 01 17:08:21 master-0-1 migrateOVN.sh[3437]: + conns='6fa43a6d-63b6-455c-b5f0-373ab8fdb460
Sep 01 17:08:21 master-0-1 migrateOVN.sh[3437]: 25456a6d-bf55-3cc5-bfa7-ac1900fd01fa
Sep 01 17:08:21 master-0-1 migrateOVN.sh[3437]: a100e09e-31ad-345e-894c-772a0676719f
Sep 01 17:08:21 master-0-1 migrateOVN.sh[3437]: 87806fe4-bec8-4826-8eca-c715d68cbf24
Sep 01 17:08:21 master-0-1 migrateOVN.sh[3437]: 76682b3f-bbe7-361b-99d0-e1dd5d7b5f6c
Sep 01 17:08:21 master-0-1 migrateOVN.sh[3437]: 937b10f3-05c3-3e07-bad6-0438b0241885
Sep 01 17:08:21 master-0-1 migrateOVN.sh[3437]: 2f9e65a4-3105-4910-86e9-04f3ddbfde79
Sep 01 17:08:21 master-0-1 migrateOVN.sh[3437]: 328d7426-4cf7-4df1-a2ff-61bdb057f653
Sep 01 17:08:21 master-0-1 migrateOVN.sh[3437]: 5dc2327c-9420-49f5-a1f7-1924f359987b
Sep 01 17:08:21 master-0-1 migrateOVN.sh[3437]: 4ddf3d3a-db51-362c-a691-d4b49b53a4de '
Sep 01 17:08:21 master-0-1 migrateOVN.sh[3437]: + IFS=
Sep 01 17:08:21 master-0-1 migrateOVN.sh[3437]: + read -r conn
Sep 01 17:08:21 master-0-1 migrateOVN.sh[3437]: ++ nmcli --get-values connection.interface-name conn show 6fa43a6d-63b6-455c-b5f0-373ab8fdb460
Sep 01 17:08:21 master-0-1 migrateOVN.sh[3437]: + new_device=br-ex
Sep 01 17:08:21 master-0-1 migrateOVN.sh[3437]: + '[' br-ex = bond0.373 ']'
Sep 01 17:08:21 master-0-1 migrateOVN.sh[3437]: + IFS=
Sep 01 17:08:21 master-0-1 migrateOVN.sh[3437]: + read -r conn
Sep 01 17:08:21 master-0-1 migrateOVN.sh[3437]: ++ nmcli --get-values connection.interface-name conn show 25456a6d-bf55-3cc5-bfa7-ac1900fd01fa
Sep 01 17:08:21 master-0-1 migrateOVN.sh[3437]: + new_device=enp0s3
Sep 01 17:08:21 master-0-1 migrateOVN.sh[3437]: + '[' enp0s3 = bond0.373 ']'
Sep 01 17:08:21 master-0-1 migrateOVN.sh[3437]: + IFS=
Sep 01 17:08:21 master-0-1 migrateOVN.sh[3437]: + read -r conn
Sep 01 17:08:21 master-0-1 migrateOVN.sh[3437]: ++ nmcli --get-values connection.interface-name conn show a100e09e-31ad-345e-894c-772a0676719f
Sep 01 17:08:21 master-0-1 migrateOVN.sh[3437]: + new_device=bond0.373
Sep 01 17:08:21 master-0-1 migrateOVN.sh[3437]: + '[' bond0.373 = bond0.373 ']'
Sep 01 17:08:21 master-0-1 migrateOVN.sh[3437]: + echo 'Detected connection a100e09e-31ad-345e-894c-772a0676719f  for device bond0.373'
Sep 01 17:08:21 master-0-1 migrateOVN.sh[3437]: Detected connection a100e09e-31ad-345e-894c-772a0676719f  for device bond0.373
Sep 01 17:08:21 master-0-1 migrateOVN.sh[3437]: ++ echo a100e09e-31ad-345e-894c-772a0676719f
Sep 01 17:08:21 master-0-1 migrateOVN.sh[3437]: ++ tr -d '[:space:]'
Sep 01 17:08:21 master-0-1 migrateOVN.sh[3437]: + detected_conn=a100e09e-31ad-345e-894c-772a0676719f
Sep 01 17:08:21 master-0-1 migrateOVN.sh[3437]: + return
Sep 01 17:08:21 master-0-1 migrateOVN.sh[3437]: + echo 'Bringing up connection a100e09e-31ad-345e-894c-772a0676719f'
Sep 01 17:08:21 master-0-1 migrateOVN.sh[3437]: Bringing up connection a100e09e-31ad-345e-894c-772a0676719f
Sep 01 17:08:21 master-0-1 migrateOVN.sh[3437]: + retries=5
Sep 01 17:08:21 master-0-1 migrateOVN.sh[3437]: + i=1
Sep 01 17:08:21 master-0-1 migrateOVN.sh[3437]: + '[' 1 -le 5 ']'
Sep 01 17:08:21 master-0-1 migrateOVN.sh[3437]: + nmcli conn down a100e09e-31ad-345e-894c-772a0676719f
Sep 01 17:08:21 master-0-1 migrateOVN.sh[3437]: Connection 'bond0.373' successfully deactivated (D-Bus active path: /org/freedesktop/NetworkManager/ActiveConnection/16)
Sep 01 17:08:21 master-0-1 migrateOVN.sh[3437]: + echo 'Brought down connection a100e09e-31ad-345e-894c-772a0676719f'
Sep 01 17:08:21 master-0-1 migrateOVN.sh[3437]: Brought down connection a100e09e-31ad-345e-894c-772a0676719f
Sep 01 17:08:21 master-0-1 migrateOVN.sh[3437]: + nmcli conn up a100e09e-31ad-345e-894c-772a0676719f
Sep 01 17:08:21 master-0-1 migrateOVN.sh[3437]: Connection successfully activated (D-Bus active path: /org/freedesktop/NetworkManager/ActiveConnection/17)
Sep 01 17:08:21 master-0-1 migrateOVN.sh[3437]: + echo 'Connection a100e09e-31ad-345e-894c-772a0676719f up successfully'
Sep 01 17:08:21 master-0-1 migrateOVN.sh[3437]: Connection a100e09e-31ad-345e-894c-772a0676719f up successfully
Sep 01 17:08:21 master-0-1 migrateOVN.sh[3437]: + break
Sep 01 17:08:21 master-0-1 migrateOVN.sh[3437]: + '[' 1 -gt 5 ']'
Sep 01 17:08:21 master-0-1 migrateOVN.sh[3437]: + sleep 10
Sep 01 17:08:31 master-0-1 migrateOVN.sh[3437]: ++ nmcli --get-values GENERAL.CONNECTION device show bond0.373
Sep 01 17:08:31 master-0-1 migrateOVN.sh[3437]: + new_conn=bond0.373
Sep 01 17:08:31 master-0-1 migrateOVN.sh[3437]: + '[' -z bond0.373 ']'
Sep 01 17:08:31 master-0-1 migrateOVN.sh[3437]: + [[ bond0.373 == \o\v\s\-\i\f\-\p\h\y\s\0 ]]
Sep 01 17:08:31 master-0-1 migrateOVN.sh[3437]: + [[ bond0.373 == \o\v\s\-\p\o\r\t\-\p\h\y\s\0 ]]
Sep 01 17:08:31 master-0-1 migrateOVN.sh[3437]: + nmcli --fields GENERAL.STATE conn show br-ex
Sep 01 17:08:31 master-0-1 migrateOVN.sh[3437]: GENERAL.STATE:                          activated
Sep 01 17:08:31 master-0-1 migrateOVN.sh[3437]: + iface_mac=52:54:00:da:93:5d
Sep 01 17:08:31 master-0-1 migrateOVN.sh[3437]: + echo 'MAC address found for iface: bond0.373: 52:54:00:da:93:5d'
Sep 01 17:08:31 master-0-1 migrateOVN.sh[3437]: MAC address found for iface: bond0.373: 52:54:00:da:93:5d
Sep 01 17:08:31 master-0-1 migrateOVN.sh[3437]: ++ ip link show bond0.373
Sep 01 17:08:31 master-0-1 migrateOVN.sh[3437]: ++ awk '{print $5; exit}'
Sep 01 17:08:31 master-0-1 migrateOVN.sh[3437]: + iface_mtu=1500
Sep 01 17:08:31 master-0-1 migrateOVN.sh[3437]: + [[ -z 1500 ]]
Sep 01 17:08:31 master-0-1 migrateOVN.sh[3437]: + echo 'MTU found for iface: bond0.373: 1500'
Sep 01 17:08:31 master-0-1 migrateOVN.sh[3437]: MTU found for iface: bond0.373: 1500
Sep 01 17:08:31 master-0-1 migrateOVN.sh[3437]: ++ nmcli --fields UUID,DEVICE conn show --active
Sep 01 17:08:31 master-0-1 migrateOVN.sh[3437]: ++ awk '/\sbond0.373\s*$/ {print $1}'
Sep 01 17:08:31 master-0-1 migrateOVN.sh[3437]: + old_conn=a100e09e-31ad-345e-894c-772a0676719f
Sep 01 17:08:31 master-0-1 migrateOVN.sh[3437]: + extra_brex_args='802-3-ethernet.mtu 1500 802-3-ethernet.cloned-mac-address 52:54:00:da:93:5d'
Sep 01 17:08:31 master-0-1 migrateOVN.sh[3437]: ++ nmcli --get-values ipv4.dhcp-client-id conn show a100e09e-31ad-345e-894c-772a0676719f
Sep 01 17:08:31 master-0-1 migrateOVN.sh[3437]: + dhcp_client_id=
Sep 01 17:08:31 master-0-1 migrateOVN.sh[3437]: + '[' -n '' ']'
Sep 01 17:08:31 master-0-1 migrateOVN.sh[3437]: + extra_brex_args+=' -ipv4.dhcp-client-id'
Sep 01 17:08:31 master-0-1 migrateOVN.sh[3437]: ++ nmcli --get-values ipv6.dhcp-duid conn show a100e09e-31ad-345e-894c-772a0676719f
Sep 01 17:08:31 master-0-1 migrateOVN.sh[3437]: + dhcp6_client_id=
Sep 01 17:08:31 master-0-1 migrateOVN.sh[3437]: + '[' -n '' ']'
Sep 01 17:08:31 master-0-1 migrateOVN.sh[3437]: + extra_brex_args+=' -ipv6.dhcp-duid'
Sep 01 17:08:31 master-0-1 migrateOVN.sh[3437]: + extra_brex_args+=' ipv4.route-metric 500 ipv6.route-metric 500'
Sep 01 17:08:31 master-0-1 migrateOVN.sh[3437]: + nmcli conn mod br-ex 802-3-ethernet.mtu 1500 802-3-ethernet.cloned-mac-address 52:54:00:da:93:5d -ipv4.dhcp-client-id -ipv6.dhcp-duid ipv4.route-metric 500 ipv6.route-metric 500
Sep 01 17:08:31 master-0-1 migrateOVN.sh[3437]: + extra_phys_args='802-3-ethernet.mtu 1500 conn.interface bond0.373 '
Sep 01 17:08:31 master-0-1 migrateOVN.sh[3437]: ++ nmcli --get-values connection.type conn show a100e09e-31ad-345e-894c-772a0676719f
Sep 01 17:08:31 master-0-1 migrateOVN.sh[3437]: + '[' vlan == vlan ']'
Sep 01 17:08:31 master-0-1 migrateOVN.sh[3437]: + iface_type=vlan
Sep 01 17:08:31 master-0-1 migrateOVN.sh[3437]: ++ nmcli --get-values vlan.id conn show a100e09e-31ad-345e-894c-772a0676719f
Sep 01 17:08:31 master-0-1 migrateOVN.sh[3437]: + vlan_id=373
Sep 01 17:08:31 master-0-1 migrateOVN.sh[3437]: + '[' -z 373 ']'
Sep 01 17:08:31 master-0-1 migrateOVN.sh[3437]: ++ nmcli --get-values vlan.parent conn show a100e09e-31ad-345e-894c-772a0676719f
Sep 01 17:08:31 master-0-1 migrateOVN.sh[3437]: + vlan_parent=bond0
Sep 01 17:08:31 master-0-1 migrateOVN.sh[3437]: + '[' -z bond0 ']'
Sep 01 17:08:31 master-0-1 migrateOVN.sh[3437]: + extra_phys_args='dev bond0 id 373'
Sep 01 17:08:31 master-0-1 migrateOVN.sh[3437]: + nmcli conn mod ovs-port-phys0 conn.interface bond0.373
Sep 01 17:08:31 master-0-1 migrateOVN.sh[3437]: ++ nmcli --get-values connection.type conn show ovs-if-phys0
Sep 01 17:08:31 master-0-1 migrateOVN.sh[3437]: + current_iface_type=bond
Sep 01 17:08:31 master-0-1 migrateOVN.sh[3437]: + '[' bond = vlan ']'
Sep 01 17:08:31 master-0-1 migrateOVN.sh[3437]: + echo 'Migration from bond to vlan detected. Will recreate ovs-if-phys0'
Sep 01 17:08:31 master-0-1 migrateOVN.sh[3437]: Migration from bond to vlan detected. Will recreate ovs-if-phys0
Sep 01 17:08:31 master-0-1 migrateOVN.sh[3437]: + nmcli conn delete ovs-if-phys0
Sep 01 17:08:31 master-0-1 migrateOVN.sh[3437]: Connection 'ovs-if-phys0' (2f9e65a4-3105-4910-86e9-04f3ddbfde79) successfully deleted.
Sep 01 17:08:31 master-0-1 migrateOVN.sh[3437]: + nmcli c add type vlan conn.interface bond0.373 master ovs-port-phys0 con-name ovs-if-phys0 connection.autoconnect-priority 100 802-3-ethernet.mtu 1500 dev bond0 id 373
Sep 01 17:08:31 master-0-1 migrateOVN.sh[3437]: Connection 'ovs-if-phys0' (7eb2a2e6-accf-4a33-8f77-94e1859a199c) successfully added.
Sep 01 17:08:32 master-0-1 migrateOVN.sh[3437]: + nmcli conn mod ovs-if-br-ex 802-3-ethernet.mtu 1500 802-3-ethernet.cloned-mac-address 52:54:00:da:93:5d ipv4.never-default yes ipv6.never-default yes
Sep 01 17:08:32 master-0-1 migrateOVN.sh[3437]: + systemctl restart NetworkManager
Sep 01 17:08:32 master-0-1 migrateOVN.sh[3437]: + sleep 5
Sep 01 17:08:37 master-0-1 migrateOVN.sh[3437]: + wait_conn ovs-if-phys0
Sep 01 17:08:37 master-0-1 migrateOVN.sh[3437]: + local conn=ovs-if-phys0
Sep 01 17:08:37 master-0-1 migrateOVN.sh[3437]: + n=0
Sep 01 17:08:37 master-0-1 migrateOVN.sh[3437]: + '[' 0 -ge 5 ']'
Sep 01 17:08:37 master-0-1 migrateOVN.sh[3437]: + nmcli c show --active
Sep 01 17:08:37 master-0-1 migrateOVN.sh[3437]: + grep ovs-if-phys0
Sep 01 17:08:37 master-0-1 migrateOVN.sh[3437]: ovs-if-phys0        7eb2a2e6-accf-4a33-8f77-94e1859a199c  vlan           bond0.373
Sep 01 17:08:37 master-0-1 migrateOVN.sh[3437]: + s=0
Sep 01 17:08:37 master-0-1 migrateOVN.sh[3437]: + break
Sep 01 17:08:37 master-0-1 migrateOVN.sh[3437]: + return 0
Sep 01 17:08:37 master-0-1 migrateOVN.sh[3437]: ++ ovs-vsctl --columns ofport --bare find interface name=bond0.373
Sep 01 17:08:37 master-0-1 migrateOVN.sh[3437]: + ofport=5
Sep 01 17:08:37 master-0-1 migrateOVN.sh[3437]: + '[' -z 5 ']'
Sep 01 17:08:37 master-0-1 migrateOVN.sh[3437]: + ovs-ofctl add-flow br-ex table=0,priority=101,in_port=LOCAL,actions=output:5
Sep 01 17:08:37 master-0-1 migrateOVN.sh[3437]: + nm-online -s
Sep 01 17:08:38 master-0-1 migrateOVN.sh[3437]: [74B blob data]
Sep 01 17:08:38 master-0-1 migrateOVN.sh[3437]: + nmcli c show
Sep 01 17:08:38 master-0-1 migrateOVN.sh[3437]: NAME                UUID                                  TYPE           DEVICE
Sep 01 17:08:38 master-0-1 migrateOVN.sh[3437]: bond0               4ddf3d3a-db51-362c-a691-d4b49b53a4de  bond           bond0
Sep 01 17:08:38 master-0-1 migrateOVN.sh[3437]: Wired connection 1  25456a6d-bf55-3cc5-bfa7-ac1900fd01fa  ethernet       enp0s3
Sep 01 17:08:38 master-0-1 migrateOVN.sh[3437]: ovs-if-br-ex        6fa43a6d-63b6-455c-b5f0-373ab8fdb460  ovs-interface  br-ex
Sep 01 17:08:38 master-0-1 migrateOVN.sh[3437]: br-ex               87806fe4-bec8-4826-8eca-c715d68cbf24  ovs-bridge     br-ex
Sep 01 17:08:38 master-0-1 migrateOVN.sh[3437]: enp0s4              76682b3f-bbe7-361b-99d0-e1dd5d7b5f6c  ethernet       enp0s4
Sep 01 17:08:38 master-0-1 migrateOVN.sh[3437]: enp0s5              937b10f3-05c3-3e07-bad6-0438b0241885  ethernet       enp0s5
Sep 01 17:08:38 master-0-1 migrateOVN.sh[3437]: ovs-if-phys0        7eb2a2e6-accf-4a33-8f77-94e1859a199c  vlan           bond0.373
Sep 01 17:08:38 master-0-1 migrateOVN.sh[3437]: ovs-port-br-ex      328d7426-4cf7-4df1-a2ff-61bdb057f653  ovs-port       br-ex
Sep 01 17:08:38 master-0-1 migrateOVN.sh[3437]: ovs-port-phys0      5dc2327c-9420-49f5-a1f7-1924f359987b  ovs-port       bond0.373
Sep 01 17:08:38 master-0-1 migrateOVN.sh[3437]: bond0.373           a100e09e-31ad-345e-894c-772a0676719f  vlan           --
Sep 01 17:08:38 master-0-1 migrateOVN.sh[3437]: ++ crictl ps --name ovnkube-node
Sep 01 17:08:38 master-0-1 migrateOVN.sh[3437]: ++ grep -v '^CONTAINER'
Sep 01 17:08:38 master-0-1 migrateOVN.sh[3437]: ++ tail -n 1
Sep 01 17:08:38 master-0-1 migrateOVN.sh[3437]: ++ awk '{print $1}'
Sep 01 17:08:40 master-0-1 migrateOVN.sh[3437]: time="2021-09-01T17:08:40Z" level=fatal msg="connect: connect endpoint 'unix:///var/run/crio/crio.sock', make sure you are running as root and the endpoint has been started: context deadline exceeded"
Sep 01 17:08:40 master-0-1 migrateOVN.sh[3437]: + container=
Sep 01 17:08:40 master-0-1 migrateOVN.sh[3437]: + '[' -z '' ']'
Sep 01 17:08:40 master-0-1 migrateOVN.sh[3437]: + echo 'WARNING: Unable to find ovnkube node container to stop'
Sep 01 17:08:40 master-0-1 migrateOVN.sh[3437]: WARNING: Unable to find ovnkube node container to stop
Sep 01 17:08:40 master-0-1 migrateOVN.sh[3437]: + copy_nm_conn_files
Sep 01 17:08:40 master-0-1 migrateOVN.sh[3437]: + src_path=/etc/NetworkManager/system-connections-merged
Sep 01 17:08:40 master-0-1 migrateOVN.sh[3437]: + dst_path=/etc/NetworkManager/system-connections
Sep 01 17:08:40 master-0-1 migrateOVN.sh[3437]: + '[' -d /etc/NetworkManager/system-connections-merged ']'
Sep 01 17:08:40 master-0-1 migrateOVN.sh[3437]: + echo '/etc/NetworkManager/system-connections-merged exists'
Sep 01 17:08:40 master-0-1 migrateOVN.sh[3437]: /etc/NetworkManager/system-connections-merged exists
Sep 01 17:08:40 master-0-1 migrateOVN.sh[3437]: ++ echo br-ex.nmconnection ovs-if-br-ex.nmconnection ovs-if-phys0.nmconnection ovs-port-phys0.nmconnection
Sep 01 17:08:40 master-0-1 migrateOVN.sh[3437]: + fileList='br-ex.nmconnection ovs-if-br-ex.nmconnection ovs-if-phys0.nmconnection ovs-port-phys0.nmconnection'
Sep 01 17:08:40 master-0-1 migrateOVN.sh[3437]: + for file in ${fileList[*]}
Sep 01 17:08:40 master-0-1 migrateOVN.sh[3437]: + '[' -f /etc/NetworkManager/system-connections-merged/br-ex.nmconnection ']'
Sep 01 17:08:40 master-0-1 migrateOVN.sh[3437]: + cp -f /etc/NetworkManager/system-connections-merged/br-ex.nmconnection /etc/NetworkManager/system-connections/br-ex.nmconnection
Sep 01 17:08:40 master-0-1 migrateOVN.sh[3437]: + for file in ${fileList[*]}
Sep 01 17:08:40 master-0-1 migrateOVN.sh[3437]: + '[' -f /etc/NetworkManager/system-connections-merged/ovs-if-br-ex.nmconnection ']'
Sep 01 17:08:40 master-0-1 migrateOVN.sh[3437]: + cp -f /etc/NetworkManager/system-connections-merged/ovs-if-br-ex.nmconnection /etc/NetworkManager/system-connections/ovs-if-br-ex.nmconnection
Sep 01 17:08:40 master-0-1 migrateOVN.sh[3437]: + for file in ${fileList[*]}
Sep 01 17:08:40 master-0-1 migrateOVN.sh[3437]: + '[' -f /etc/NetworkManager/system-connections-merged/ovs-if-phys0.nmconnection ']'
Sep 01 17:08:40 master-0-1 migrateOVN.sh[3437]: + cp -f /etc/NetworkManager/system-connections-merged/ovs-if-phys0.nmconnection /etc/NetworkManager/system-connections/ovs-if-phys0.nmconnection
Sep 01 17:08:40 master-0-1 migrateOVN.sh[3437]: + for file in ${fileList[*]}
Sep 01 17:08:40 master-0-1 migrateOVN.sh[3437]: + '[' -f /etc/NetworkManager/system-connections-merged/ovs-port-phys0.nmconnection ']'
Sep 01 17:08:40 master-0-1 migrateOVN.sh[3437]: + cp -f /etc/NetworkManager/system-connections-merged/ovs-port-phys0.nmconnection /etc/NetworkManager/system-connections/ovs-port-phys0.nmconnection
Sep 01 17:08:40 master-0-1 migrateOVN.sh[3437]: + echo 'OVS Migration has completed updating Network Manager connections!'
Sep 01 17:08:40 master-0-1 migrateOVN.sh[3437]: OVS Migration has completed updating Network Manager connections!
Sep 01 17:08:40 master-0-1 migrateOVN.sh[3437]: + revert 0
Sep 01 17:08:40 master-0-1 migrateOVN.sh[3437]: + '[' 0 -gt 0 ']'
Sep 01 17:08:40 master-0-1 systemd[1]: ovs-migration.service: Succeeded.
Sep 01 17:08:40 master-0-1 systemd[1]: Started Migrates OVS configuration to use a new interface on the host.
Sep 01 17:08:40 master-0-1 systemd[1]: ovs-migration.service: Consumed 1.228s CPU time
```